storyblok/storyblok-design-system
# Refactor SbIcon component to SFC

## Pull request type

Jira Link: NA




## How to test this PR

Please navigate to the [SbIcon page](http://localhost:6006/?path=/docs/design-system-components-sbicon--default) in Storybook and verify that the icon component behaves as expected:

- Each prop listed should have the expected effect on the rendered icon, when changed.
- For any instance of SbIcon on this page, and any other page (i.e. SbButton), the component should display the correct icon and styling.

## What is the new behavior?



- SBIcon component is refactored to SbIcon.vue.
- A new unit test, 'when pass role attribute', was added to src/components/Icon/__tests__/icon.spec.js, and the result of that test should be visible when running `yarn test:unit`

## Other information

Since this is a very simple component, I consolidated as much as possible into the .vue file. I left the utils.js file because the exported iconSizes array is used both by the .vue file and SbIcon's Storybook config file.
